src/main.rs

```diff
@@ -53,6 +53,25 @@ const CARGO: &'static str = "cargo";
 const MIN_DEPTH: usize = 1;
 const MAX_DEPTH: usize = 1;
 
+fn generate_cargo_cmd(path: &PathBuf, commands: &Vec<String>) -> Command {
+
+    let mut cargo_cmd = Command::new(CARGO);
+
+    // Take a clone of the commands so that the manifest can be passed to the
+    // cargo command, this is to any references to files in the output are relative
+    // to the current directory.
+    let mut commands = commands.clone();
+
+    commands.insert(1, "--manifest-path".to_string());
+    commands.insert(2, format!("{}/Cargo.toml", path.to_string_lossy()));
+
+    for arg in commands {
+        cargo_cmd.arg(arg);
+    }
```

Comment:
Actually, can't you just do:

```
cargo_cmd,arg("--manifest-path");
cargo_cmd.arg(...path sruff...);
for arg in &commands {
    cargo_cmd.arg(arg);
}
```

Then you don't need to own `commands` at all.